src/runtime_src/driver/zynq/user/shim.cpp

```diff
@@ -143,144 +126,160 @@ ZYNQShim::~ZYNQShim()
 }
 #endif
 
+void ZYNQShim::mapKernelControl(std::vector<uint64_t> cus_offset) {
+  void *ptr = NULL;
+  if (cus_offset.size() == 0) {
+    return;
+  }
+
+  auto cu = cus_offset.begin();
 
-//TODO: UKP: return definition is not defined.
-size_t ZYNQShim::xclWrite(xclAddressSpace space, uint64_t offset, const void *hostBuf, size_t size)
-{
-  if(!hostBuf) {
+  while (cu != cus_offset.end()) {
```

Comment:
Capture the end outside of while loop or else end() is reevaluated in every iteration.
cuEnd = cus_offset.end()
while (cu != cuEnd) {

Reply:
Got it.